### PR TITLE
[workbench] modular configuration for genesis

### PR DESCRIPTION
**Pull request by andreabedini**
# Description

This introduces a new way to configure the genesis creation, based on
NixOS modules (see [1] and [2] for reference). This replaces some of
the complex jq-based logic present in the workbench.

The new code path is enabled only if either
- the environment variable WB_MODULAR_GENESIS is set to 1
- the option --enable-modular-genesis is passed to `wb`

Workbench changes:
- Introduce configuration modules in `nix/workbench/modules`
- Add `wb evaluate` command to evaluate the configuration

[1] https://nix.dev/tutorials/module-system/module-system
[2] https://nixos.org/manual/nixos/stable/#sec-writing-modules

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - roundtrip tests
  - integration tests
  See [Runnings tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Any changes are noted in the `CHANGELOG.md` for affected package
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] Self-reviewed the diff

# Note on CI
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges.  Please contact IOG node developers to do this
for you.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -71,3 +71,6 @@ cardano-tracer/cardano-tracer-test
 
 # IntellIJ project folder
 .idea/
+
+# Workbench
+run/
```
Comment (mgmeier) [inline .gitignore:76]: Redundant - pls remove

**Review by mgmeier — COMMENTED** [review]
Thank you so much @andreabedini . This is excellent work.

File: .gitignore (hunk 1 above)
Comment (andreabedini) [reply .gitignore]: :+1:

**Review by mgmeier — APPROVED** [review]
LGTM - thank you so much @andreabedini

**Pull request merged**